### ISSUE TITLE
Subclasses of XmlLexer that deepcopy tokens cannot output LaTeX

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#875]
(Original issue 1168 created by nwhetsell on 2015-11-13T18:51:47.133115+00:00)

A subclass of `XmlLexer` that copies `XmlLexer`’s tokens, like

```py
import copy
from pygments.lexers.html import XmlLexer
__all__ = ['DummyLexer']
class DummyLexer(XmlLexer):
    name = 'Dummy'
    aliases = ['dummy']
    filenames = ['*.dummy']
    mimetypes = []
    tokens = copy.deepcopy(XmlLexer.tokens)
```

throws an exception when formatting output as LaTeX. The command I’m using is

```
./pygmentize -v -l dummy -f latex test.dummy
```

and its output is

```
\begin{Verbatim}[commandchars=\\\{\}]

*****************************************************************
An unhandled exception occurred while highlighting.
Please report the whole traceback to the issue tracker at
<https://bitbucket.org/birkenfeld/pygments-main/issues>.
*****************************************************************

Traceback (most recent call last):
  File "./pygmentize", line 6, in <module>
    sys.exit(pygments.cmdline.main(sys.argv))
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/cmdline.py", line 504, in main
    return main_inner(popts, args, usage)
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/cmdline.py", line 468, in main_inner
    highlight(code, lexer, fmter, outfile)
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/__init__.py", line 87, in highlight
    return format(lex(code, lexer), formatter, outfile)
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/__init__.py", line 69, in format
    formatter.format(tokens, outfile)
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/formatter.py", line 95, in format
    return self.format_unencoded(tokensource, outfile)
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/formatters/latex.py", line 399, in format_unencoded
    styles.append(_get_ttype_name(ttype))
  File "/Users/nathanwhetsell/Documents/pygments-main/pygments/formatters/latex.py", line 143, in _get_ttype_name
    aname = ttype[-1] + aname
TypeError: 'NoneType' object has no attribute '__getitem__'
```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2016-02-02T12:03:11.236448+00:00)

Thanks for the report!

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2016-02-02T12:03:30.101638+00:00)

Fixed deepcopy-ing of Token instances (closes #1168)

→ <<cset 0ddfc6cf8658>>
